runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -505,24 +505,43 @@ jvmtiNotifyFramePop(jvmtiEnv* env,
 		ENSURE_NON_NEGATIVE(depth);
 
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
-		if (rc == JVMTI_ERROR_NONE) {
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
-			if ((currentThread == targetThread) || (targetThread->publicFlags & J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND))  {
-				J9StackWalkState walkState;
+		if (JVMTI_ERROR_NONE == rc) {
+#if JAVA_SPEC_VERSION >= 19
+			if (NULL != targetThread)
+#endif /* JAVA_SPEC_VERSION >= 19 */
+			{
+				vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
+			}
+			if ((currentThread == targetThread)
+			|| OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND)
```

Comment:
Draft. Modify as needed for accuracy and neatness.
```suggestion
			BOOLEAN isVThreadSuspended = FALSE;
			//halthaltThreadForInspection
			if ((NULL != thread) && (NULL == targetThread)) {
				//assert in getVMThread will assure that this is a virtual thread
				jint vthreadState = J9VMJAVALANGVIRTUALTHREAD_STATE(currentThread, J9_JNI_UNWRAP_REFERENCE(thread));
				isVThreadSuspended = OMR_ARE_ANY_BITS_SET(vthreadState, JVMTI_VTHREAD_STATE_SUSPENDED);
			}

			if ((currentThread == targetThread)
			|| isVThreadSuspended
			|| ((NULL != targetThread) && OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND))
```